Indexer/CorpusIndexingCommand.php

```diff
@@ -158,10 +158,16 @@ public function __invoke()
         }
         $subjects = $wrappingIterator;
         $updateQuery = $this->getSolariumClient()->createUpdate();
+
         //initially delete all indexes - todo allow disambiguation between types of document
         if ($this->shouldPreDelete) {
```

Comment:
Don't we just need to change this line to

```php
if ($this->shouldPreDelete || !is_null($this->deleteQuery)) {
```

i.e. if the delete query has been actively set, set it? I think that should be sufficient. Having two different conditionals that can't really both be true seems confusing.